spec/system/user_index_spec.rb

```diff
@@ -0,0 +1,21 @@
+# user_index_spec.rb
+
+require 'rails_helper'
+
+RSpec.feature 'User Index Page', type: :feature do
+  before do
+    # Assuming @users is populated with test data or using FactoryBot
+    @users = FactoryBot.create_list(:user, 3)
+    visit users_path
+  end
+
+  scenario 'displays user information' do
+    within '.users' do
+      @users.each do |user|
+        expect(page).to have_css('.user-name', text: user.name)
+        expect(page).to have_css("img[src='#{user.photo}'][alt='#{user.name}']")
+        expect(page).to have_content("Number of posts: #{user.posts_counter}")
```

Comment:
- Hello, You have a good job so far. I would suggest you add this test you left behind: **Tests that redirects to the user's show page when you click on a user**. I am requesting this because you need to test all stories provided (Project requirements)